core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1150,7 +1151,17 @@ object SnappyContext extends Logging {
       // clear current hive catalog connection
       Hive.closeCurrent()
       if (ExternalStoreUtils.isLocalMode(sc)) {
-        ServiceUtils.invokeStopFabricServer(sc)
+        val user = sc.getConf.get(Constant.STORE_PROPERTY_PREFIX + Attribute.USERNAME_ATTR, "")
+        val props = if (!user.isEmpty) {
```

Comment:
cleaner way is pattern matching:

sc.conf.getOption(Constant.STORE_PROPERTY_PREFIX + Attribute.USERNAME_ATTR) match {
  case Some(user) => ...
  case None => null
}

Reply:
@ashetkar
Also I see that explicit "Constant.STORE_PROPERTY_PREFIX + Attribute.USERNAME_ATTR" has been used in all places for security user/password checks. For smart connector the prefix changes to "spark.snappydata.store" which is also checked I think.

This should instead be added as property in io.snappydata.Property (Literals.scala) like other store properties there. Then: Property.User.getOption(...) which will automatically check for both prefixes (snappydata.store and spark.snappydata.store). Need not be part of these changes but I think this cleanup should be done at some point.